ptbcontrib/ptb_sqlalchemy_jobstore/jobstore.py

```diff
@@ -101,15 +113,24 @@ def _reconstitute_job(self, job_state: bytes) -> APSJob:
         Args:
             job_state (:obj:`str`): String containing pickled job state.
         """
-        job = super()._reconstitute_job(job_state)  # pylint: disable=W0212
+        job: APSJob = super()._reconstitute_job(job_state)  # pylint: disable=W0212
 
         # Here we rebuild callback context for the job which
         # are going for execution.
-        tg_job = telegram.ext.Job(
-            callback=None,
-            name=job.args[0],
-            context=job.args[1],
+        name, data, chat_id, user_id, module = job.args
+        callback_module = import_module(module)
+        tg_job = Job(
+            callback=getattr(callback_module, name),
+            chat_id=chat_id,
+            user_id=user_id,
+            name=name,
+            data=data,
+        )
+        ctx = CallbackContext.from_job(tg_job, self.application)
+        job._modify(  # pylint: disable=W0212
+            args=(
+                ctx.job,
+                self.application,
+            )
```

Comment:
```suggestion
            args=(self.application,)
```
In v20 the only argument for the job is the application itself. This should also simplify some of the above logic

Reply:
When passing only application it still requires a second parameter which is application, hmm.

Reply:
Can you show the error that you get, including the traceback?

Reply:
```
job._modify(args=(self.application,))
Traceback (most recent call last):
  File "telegram_bot/venv/lib/python3.10/site-packages/apscheduler/job.py", line 180, in _modify
    check_callable_args(func, args, kwargs)
  File "telegram_bot/venv/lib/python3.10/site-packages/apscheduler/util.py", line 391, in check_callable_args
    raise ValueError('The following arguments have not been supplied: %s' %
ValueError: The following arguments have not been supplied: application
```

Reply:
Okay the problem is that serializing the job turns a *bound* method `some_ptb_job.run` into the *function* `tg.ext.Job.run`, which we need to pass the `self` argument to.